main.go

```diff
@@ -30,9 +30,21 @@ func main() {
 			os.Exit(1)
 		}
 	}()
-
 	app := NewTGFApplication(os.Args[1:])
+	os.Exit(runTgf(app))
+}
+
+func runTgf(app *TGFApplication) int {
```

Comment:
We made a lot of effort to make the application more object oriented. It's a good idea to split the main function and make it testable, but you should avoid passing app & config as parameter.

Reply:
Okay sounds good.

But why is it a bad idea ?

Reply:
Make it more testable and mockable is certainly a good idea.

The problem is that we no not need to supply app & config since config already contains the app. Make it object oriented actually just move the parameter. `run(param)` vs `param.run()`.